cedar-policy/src/api/err.rs

```diff
@@ -564,26 +641,28 @@ pub enum PolicySetError {
     #[error("Error serializing a policy to JSON: {0}")]
     #[diagnostic(transparent)]
     ToJson(#[from] PolicyToJsonError),
-    /// Errors encountered in JSON ser/de
-    #[error("Error serializing or deserializing from JSON: {0})")]
+    /// Errors encountered in JSON ser/de of the policy set (as opposed to individual policies)
+    #[error("Error serializing / deserializing PolicySet to / from JSON: {0})")]
```

Comment:
Not sure if we want to implement `Diagnostic` for this. IMO `serde_json::Error` is probably better for this case because we'll automatically get improvements to `serde_json` error reporting. The main downside is it's less consistent. (Note, if I understand the code correctly this will only happen if the `PolicySet` isn't in our expected JSON format. We should never fail `PolicySet -> JSON`.)

Reply:
Leaving as `serde_json::Error for now at least.